### PR TITLE
DocFX sanity check

### PR DESCRIPTION
**Pull request by reyang**
## Changes

Added DocFx sanity check to the CI.
Fixed all the broken links.
Once this PR gets merged, no one is going to introduce broken link again.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +22,6 @@ The OpenTelemetry specification describes the cross-language requirements and ex
   - [Resource](specification/resource/sdk.md)
   - [Configuration](specification/sdk-configuration.md)
 - Data Specification
-  - [Semantic Conventions](specification/overview.md#semantic-conventions.md)
```
Comment (iNikem) [inline README.md:26]: This should not be removed, but replaced with https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#semantic-conventions

File: .github/workflows/docfx.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  build:
+    runs-on: windows-latest
+
+    steps:
+    - name: check out code
+      uses: actions/checkout@v2
+
+    - name: install docfx
```
Comment (iNikem) [inline .github/workflows/docfx.yml:15]: Why do you install it, if it is already checked into source control?

Comment (reyang) [reply .github/workflows/docfx.yml:15]: The tool is not checked into source control.

Comment (iNikem) [reply .github/workflows/docfx.yml:15]: What is `docfx.cmd` file below then?

Comment (Oberon00) [reply .github/workflows/docfx.yml:15]: A wrapper script to run docfx with the right options & exit code? https://github.com/open-telemetry/opentelemetry-specification/pull/742/files#diff-2fa7a6d21c1369efbff0a5d848afef7a

Comment (reyang) [reply .github/workflows/docfx.yml:15]: Yes @Oberon00 is correct, it is a wrapper script to run `docfx.exe`.

**Pull request merged**